lib/llvm/execution_engine.rb

```diff
@@ -182,7 +186,9 @@ class MCJITCompiler < ExecutionEngine
     def initialize(mod, options = {})
       options = {
         :opt_level             => 2, # LLVMCodeGenLevelDefault
-        :code_model            => 0, # LLVMCodeModelDefault
+        # code_model causes segfault with LLVMCodeModelDefault (0) so using
+        # LLVMCodeModelJITDefault (1) instead
+        :code_model            => 1,
```

Comment:
-- Ruby level backtrace information ----------------------------------------
/var/lib/gems/2.3.0/gems/minitest-5.10.1/lib/minitest.rb:62:in `block in autorun'
/var/lib/gems/2.3.0/gems/minitest-5.10.1/lib/minitest.rb:135:in `run'
/var/lib/gems/2.3.0/gems/minitest-5.10.1/lib/minitest.rb:158:in `__run'
/var/lib/gems/2.3.0/gems/minitest-5.10.1/lib/minitest.rb:158:in `map'
/var/lib/gems/2.3.0/gems/minitest-5.10.1/lib/minitest.rb:158:in `block in __run'
/var/lib/gems/2.3.0/gems/minitest-5.10.1/lib/minitest.rb:308:in `run'
/var/lib/gems/2.3.0/gems/minitest-5.10.1/lib/minitest.rb:336:in `with_info_handler'
/var/lib/gems/2.3.0/gems/minitest-5.10.1/lib/minitest.rb:349:in `on_signal'
/var/lib/gems/2.3.0/gems/minitest-5.10.1/lib/minitest.rb:309:in `block in run'
/var/lib/gems/2.3.0/gems/minitest-5.10.1/lib/minitest.rb:309:in `each'
/var/lib/gems/2.3.0/gems/minitest-5.10.1/lib/minitest.rb:310:in `block (2 levels) in run'
/var/lib/gems/2.3.0/gems/minitest-5.10.1/lib/minitest.rb:323:in `run_one_method'
/var/lib/gems/2.3.0/gems/minitest-5.10.1/lib/minitest.rb:822:in `run_one_method'
/var/lib/gems/2.3.0/gems/minitest-reporters-1.1.14/lib/minitest/reporters.rb:48:in `run_with_hooks'
/var/lib/gems/2.3.0/gems/minitest-5.10.1/lib/minitest/test.rb:100:in `run'
/var/lib/gems/2.3.0/gems/minitest-5.10.1/lib/minitest/test.rb:273:in `with_info_handler'
/var/lib/gems/2.3.0/gems/minitest-5.10.1/lib/minitest.rb:349:in `on_signal'
/var/lib/gems/2.3.0/gems/minitest-5.10.1/lib/minitest/test.rb:101:in `block in run'
/var/lib/gems/2.3.0/gems/minitest-5.10.1/lib/minitest/test.rb:253:in `time_it'
/var/lib/gems/2.3.0/gems/minitest-5.10.1/lib/minitest/test.rb:102:in `block (2 levels) in run'
/var/lib/gems/2.3.0/gems/minitest-5.10.1/lib/minitest/test.rb:202:in `capture_exceptions'
/var/lib/gems/2.3.0/gems/minitest-5.10.1/lib/minitest/test.rb:105:in `block (3 levels) in run'
[snip]/ruby-llvm/test/call_test.rb:79:in `test_external_string'
[snip]/ruby-llvm/test/test_helper.rb:53:in `run_function_on_module'
[snip]/ruby-llvm/lib/llvm/execution_engine.rb:221:in `run_function'
[snip]/ruby-llvm/lib/llvm/execution_engine.rb:221:in `call'

-- C level backtrace information -------------------------------------------
/usr/lib/x86_64-linux-gnu/libruby-2.3.so.2.3 [0x7fc34ef18ca5]
/usr/lib/x86_64-linux-gnu/libruby-2.3.so.2.3 [0x7fc34ef18edc]
/usr/lib/x86_64-linux-gnu/libruby-2.3.so.2.3 [0x7fc34edf2944]
/usr/lib/x86_64-linux-gnu/libruby-2.3.so.2.3 [0x7fc34eea4c3e]
/lib/x86_64-linux-gnu/libc.so.6 [0x7fc34e9fa4b0]
/lib/x86_64-linux-gnu/libc.so.6(strlen+0x26) [0x7fc34ea4fb96]
/lib/x86_64-linux-gnu/libc.so.6(getenv+0x16) [0x7fc34e9fe786]
[0x7fc34f41b00b]
/usr/lib/x86_64-linux-gnu/libffi.so.6(ffi_call_unix64+0x4c) [0x7fc34a51ae40]
/usr/lib/x86_64-linux-gnu/libffi.so.6(ffi_call+0x2eb) [0x7fc34a51a8ab]
/var/lib/gems/2.3.0/gems/ffi-1.9.18/lib/ffi_c.so(rbffi_CallFunction+0xf0) [0x7fc34a72a3b0] Call.c:430
/usr/lib/x86_64-linux-gnu/libruby-2.3.so.2.3 [0x7fc34ef0650b]
/usr/lib/x86_64-linux-gnu/libruby-2.3.so.2.3 [0x7fc34ef0a269]
/usr/lib/x86_64-linux-gnu/libruby-2.3.so.2.3 [0x7fc34ef0f142]
/usr/lib/x86_64-linux-gnu/libruby-2.3.so.2.3(rb_yield+0x364) [0x7fc34ef15cb4]
/usr/lib/x86_64-linux-gnu/libruby-2.3.so.2.3(rb_ary_each+0x3d) [0x7fc34edb3e4d]
/usr/lib/x86_64-linux-gnu/libruby-2.3.so.2.3 [0x7fc34ef0650b]
/usr/lib/x86_64-linux-gnu/libruby-2.3.so.2.3 [0x7fc34ef0a19c]
/usr/lib/x86_64-linux-gnu/libruby-2.3.so.2.3 [0x7fc34ef0f142]
/usr/lib/x86_64-linux-gnu/libruby-2.3.so.2.3(rb_yield+0x364) [0x7fc34ef15cb4]
/usr/lib/x86_64-linux-gnu/libruby-2.3.so.2.3 [0x7fc34edb7a4d]
/usr/lib/x86_64-linux-gnu/libruby-2.3.so.2.3 [0x7fc34ef0650b]
/usr/lib/x86_64-linux-gnu/libruby-2.3.so.2.3 [0x7fc34ef144a3]
/usr/lib/x86_64-linux-gnu/libruby-2.3.so.2.3 [0x7fc34ef154d3]
/usr/lib/x86_64-linux-gnu/libruby-2.3.so.2.3 [0x7fc34ef0a19c]
/usr/lib/x86_64-linux-gnu/libruby-2.3.so.2.3 [0x7fc34ef0f142]
/usr/lib/x86_64-linux-gnu/libruby-2.3.so.2.3 [0x7fc34ef0fd35]
/usr/lib/x86_64-linux-gnu/libruby-2.3.so.2.3 [0x7fc34ef0fe13]
/usr/lib/x86_64-linux-gnu/libruby-2.3.so.2.3(rb_proc_call+0x42) [0x7fc34edfeb12]
/usr/lib/x86_64-linux-gnu/libruby-2.3.so.2.3 [0x7fc34edf5c24]
/usr/lib/x86_64-linux-gnu/libruby-2.3.so.2.3(rb_exec_end_proc+0xbb) [0x7fc34edf8f4b]
/usr/lib/x86_64-linux-gnu/libruby-2.3.so.2.3 [0x7fc34edf9073]
/usr/lib/x86_64-linux-gnu/libruby-2.3.so.2.3(ruby_cleanup+0x1fa) [0x7fc34edf941a]
/usr/lib/x86_64-linux-gnu/libruby-2.3.so.2.3(ruby_run_node+0x25) [0x7fc34edf9845]
/usr/bin/ruby2.3 [0x40087b]
/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf0) [0x7fc34e9e5830]
/usr/bin/ruby2.3(_start+0x29) [0x4008a9]